packages/pouchdb-adapter-utils/src/preprocessAttachments.js

```diff
@@ -1,72 +1,81 @@
 import {
   atob,
   btoa,
-  arrayBufferToBinaryString as arrayBuffToBinString,
-  readAsArrayBuffer,
   binaryStringToBlobOrBuffer as binStringToBlobOrBuffer,
-  arrayBufferToBase64 as arrayBuffToB64
+  blobOrBufferToBinaryString as blufferToBinaryString,
+  blobOrBufferToBase64 as blufferToBase64
 } from 'pouchdb-binary-utils';
 import { createError, BAD_ARG } from 'pouchdb-errors';
 import { binaryMd5 } from 'pouchdb-md5';
 
-function preprocessAttachments(docInfos, blobType, callback) {
-
-  if (!docInfos.length) {
-    return callback();
+function parseBase64(data) {
+  try {
+    return atob(data);
+  } catch (e) {
+    var err = createError(BAD_ARG,
+      'Attachment is not a valid base64 string');
+    return {error: err};
   }
+}
 
-  var docv = 0;
-
-  function parseBase64(data) {
-    try {
-      return atob(data);
-    } catch (e) {
-      var err = createError(BAD_ARG,
-        'Attachment is not a valid base64 string');
-      return {error: err};
-    }
+function preprocessString(att, blobType, callback) {
+  var asBinary = parseBase64(att.data);
+  if (asBinary.error) {
+    return callback(asBinary.error);
   }
 
-  function preprocessAttachment(att, callback) {
-    if (att.stub) {
-      return callback();
-    }
-    if (typeof att.data === 'string') {
-      // input is assumed to be a base64 string
-
-      var asBinary = parseBase64(att.data);
-      if (asBinary.error) {
-        return callback(asBinary.error);
-      }
+  att.length = asBinary.length;
+  if (blobType === 'blob') {
+    att.data = binStringToBlobOrBuffer(asBinary, att.content_type);
+  } else if (blobType === 'base64') {
+    att.data = btoa(asBinary);
+  } else { // binary
+    att.data = asBinary;
+  }
+  binaryMd5(asBinary, function (result) {
+    att.digest = 'md5-' + result;
+    callback();
+  });
+}
 
-      att.length = asBinary.length;
-      if (blobType === 'blob') {
-        att.data = binStringToBlobOrBuffer(asBinary, att.content_type);
-      } else if (blobType === 'base64') {
-        att.data = btoa(asBinary);
-      } else { // binary
-        att.data = asBinary;
-      }
-      binaryMd5(asBinary, function (result) {
-        att.digest = 'md5-' + result;
+function preprocessBlob(att, blobType, callback) {
+  binaryMd5(att.data, function (md5) {
```

Comment:
instead of taking a string or an ArrayBuffer as input, `binaryMd5` now takes a string or a Blob as input. It should never ever take an ArrayBuffer as input, because we should never be passing it a full ArrayBuffer.